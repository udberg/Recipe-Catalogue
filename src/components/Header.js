import classNames from 'classnames';
import { Link } from 'react-router-dom';
import Search from '../containers/Search';
import styles from '../styles/Header.module.css';

const cx = classNames;

const Header = () => (
  <nav className={cx(styles.nav)}>
    <ul className={cx(styles.menus)}>
      <li className={cx(styles.menu__item)}>
        <Link className={cx(styles.menu__link)} to="/">Home</Link>
      </li>
      <li className={cx(styles.menu__item)}>
        <Link className={cx(styles.menu__link)} to="/categories">Categories</Link>
      </li>
    </ul>

    <div className={cx(styles.menu__search)}>
      <Search />
    </div>
  </nav>
);

export default Header;
