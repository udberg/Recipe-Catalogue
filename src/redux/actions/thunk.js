import { fetchMeals, fetchMeal, fetchSearchMeals } from '../../API/api';
import {
  loadingMeal, errorMeal, successMeal, loadingMeals, errorMeals, successMeals,
} from './index';

const fetchMealsByCategory = (category) => async (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchMeals(category);
  try {
    const data = await response;
    return dispatch(successMeals(data.meals));
  } catch (e) {
    return dispatch(errorMeals('Error while fetching data.'));
  }
};

const searchMeals = (query) => async (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchSearchMeals(query);
  try {
    const data = await response;
    if (data.meals) {
      return dispatch(successMeals(data.meals));
    }
    return dispatch(errorMeals(`There is no meal with name: ${query}`));
  } catch (e) {
    return dispatch(errorMeals('Error while fetching data.'));
  }
};

const fetchMealById = (id) => async (dispatch) => {
  dispatch(loadingMeal());
  const response = fetchMeal(id);
  try {
    const data = await response;
    return dispatch(successMeal(data.meals[0]));
  } catch (e) {
    return dispatch(errorMeal('Error while fetching data.'));
  }
};

export { fetchMealsByCategory, searchMeals, fetchMealById };
