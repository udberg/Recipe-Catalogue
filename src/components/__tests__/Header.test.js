import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import ConfigureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Header from '../Header';

const mockStore = ConfigureMockStore([thunk]);

let store;

beforeEach(() => {
  store = mockStore({});
});

test('should match the snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
