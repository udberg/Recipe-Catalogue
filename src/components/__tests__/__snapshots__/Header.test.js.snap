// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<nav
  class="nav"
>
  <ul
    class="menus"
  >
    <li
      class="menu__item"
    >
      <a
        class="menu__link"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="menu__item"
    >
      <a
        class="menu__link"
        href="/categories"
      >
        Categories
      </a>
    </li>
  </ul>
  <div
    class="menu__search"
  >
    <form>
      <input
        class="search-input"
        name="search"
        placeholder="Search your recipes"
        type="search"
      />
    </form>
  </div>
</nav>
`;
